addons/interaction/functions/fnc_takeLauncher.sqf

```diff
@@ -0,0 +1,46 @@
+#include "..\script_component.hpp"
+/*
+ * Author: Timi007. Cplhardcore
+ * Pass launcher to another unit.
+ *
+ * Arguments:
+ * 0: Unit that passes the launcher <OBJECT>
+ * 1: Unit to pass the launcher to <OBJECT>
+ * 2: launcher classname <STRING>
+ * 3: Play passing animation <BOOL> (default: true)
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [_player, _target, "launch_NLAW_F"] call ace_interaction_fnc_passLauncher
+ *
+ * Public: No
+ */
+
+params ["_player", "_target", "_launcher", ["_animate", true, [true]]];
+TRACE_4("Pass launcher params",_player,_target,_launcher,_animate);
+
+if (_launcher isEqualTo "") exitWith {ERROR("No launcher specified.")};
+private _hasLauncher = secondaryWeapon _player;
+if (_hasLauncher == "") exitWith {ERROR("Cannot add launcher to target due to lack of inventory space.")};
+
+private _CfgWeapons = configFile >> "CfgWeapons" >> _launcher;
+private _attachments = _target weaponAccessories _launcher;
+private _launcherAmmo = (currentMagazine _unit) select {
+    _x in getArray (configFile >> "CfgWeapons" >> _launcherClassname >> "magazines")
+};
```

Comment:
1) `_unit` is undefined.
2) `currentMagazine` is a string. I'm guessing you wanted `magazines`?
3) Ammo counts are disregarded, when they need to be taken into account (some Halo launchers use magazines with multiple rounds).
4) This code disregards any compatible ammo added via magwells.

I'm honestly not sure what you're trying to achieve here. Are you trying to just pass the loaded magazine or are you trying to pass all compatible magazines for the launcher?

Reply:
Trying to get the currently loaded magazine in the launcher, but from what I can tell there isn't a command to do that, so I'm looking at using magazines

Reply:
There is, `currentMagazineDetails`

Reply:
Don't use `currentMagazineDetail`, you have to do parsing to get the ammo count.
`weaponState` gives you both magazine classname and ammo count.

Reply:
I said the same thing on discord and Drofresh said that, completely forgot about weaponState again